tony-core/src/main/java/com/linkedin/tony/ApplicationMaster.java

```diff
@@ -1074,10 +1076,15 @@ public void onContainersAllocated(List<Container> containers) {
       for (Container container : containers) {
         // Need to explicitly remove container requests from remoteRequestsTable in AMRMClient, otherwise
         // resources get double-requested (YARN-1902)
+        int numGPU = 0;
+        if (ResourceUtils.getResourceTypeIndex().containsKey(Constants.GPU_URI)) {
+          numGPU = (int) container.getResource().getResourceInformation(Constants.GPU_URI).getValue();
+        }
+
         amRMClient.removeContainerRequest(Utils.setupContainerRequestForRM(new JobContainerRequest(
             "", 1, container.getResource().getMemorySize(),
             container.getResource().getVirtualCores(),
-            (int) container.getResource().getResourceInformation(Constants.GPU_URI).getValue(),
+            numGPU,
```

Comment:
Is there a unit-test that ensures that this codepath works? We need one if there isn't one. We need to test both cases(if(true) and if(false)).